### PR TITLE
Change return type of ListTaskRunsResult

### PR DESCRIPTION
**Pull request by yuege01**
Change return type of ListTaskRunsResult to a list of TaskRunResults instead of TaskRuns

Modified api.proto for ListTaskRunsResponse

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by yuege01** [comment]
@wlynch @dibyom

File: results/cmd/api/main_test.go [hunk 1]
```diff
@@ -555,7 +555,10 @@ func TestListTaskRuns(t *testing.T) {
 			if tc.expectStatus != status.Code(err) {
 				t.Fatalf("failed on test %v: %v", tc.name, err)
 			}
-			gotList := res.GetItems()
+			var gotList []*pb.TaskRun
+			for _, taskrunResult := range res.GetItems() {
+				gotList = append(gotList, taskrunResult.GetTaskRun())
+			}
```
Comment (wlynch) [inline results/cmd/api/main_test.go:561]: Only looking at the taskruns means that we are ignoring the extra data being returned by TaskRunResults. We should modify the expected results to include the TaskRunResults so we can verify that it's being set properly.

We'll need a way to optionally make ID values deterministic for testing (easiest thing to do is likely just making the ID generator func configurable by the package, and setting it to something deterministic in tests).

Comment (yuege01) [reply results/cmd/api/main_test.go]: done

File: results/cmd/api/main.go [hunk 2]
```diff
@@ -65,6 +65,7 @@ type server struct {
 	env *cel.Env
 	db  *sql.DB
 }
+type newID func() string
```
Comment (wlynch) [inline results/cmd/api/main.go:68]: Doesn't look like this is being used. Remove.

Comment (yuege01) [reply results/cmd/api/main.go]: done

**Pull request merged**